mantidproject/mantid
# Reflectometry GUI restructuring 5: presenter communication

**Description of work.**
Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

This PR adds cross-presenter communication:
- The Instrument combo boxes are now synchronised between the main `Runs` tab and its child `RunsTable` component.
- Inputs that would affect the reduction are disabled during processing. This requires cross-presenter communication to enable/disable widgets across several different tabs when processing is started or stopped.
- The Pause buttons are enabled during processing.

**Report to:** nobody

**To test:**

- Open the `ISIS Reflectometry` GUI
- Change either of the two `Instrument` combo boxes and the other should change in sync
- Click `Autoprocess`. The Pause buttons should become enabled, and the following inputs should be greyed out:
  - The autoprocess and process buttons
  - The toolbar buttons to edit table rows
  - The instrument, investigation and Search inputs
  - inputs on the Event, Experiment and Instrument settings tabs
  - the `Save as ASCII on completion` check box on the Save tab
- Toggle to another Batch with the tabs on the left and check that this only affects the current Batch.
- Click `Pause` and things should reset to their initial state.
- Click `Process`. This should have the same effect as `Autoprocess` except that the Investigation Id entry and Search button are not disabled. This is because these inputs are not required when the user is manually processing runs so they can be changed.

Fixes #23041 

*This does not require release notes* because it only concerns under-the-hood changes. There is no change in behaviour compared to the existing reflectometry GUI.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
